pkg/util/clusteroperator/status.go

```diff
@@ -0,0 +1,65 @@
+package clusteroperator
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetStatusCondition returns the result of setting the specified condition in
+// the given slice of conditions.
+func SetStatusCondition(oldConditions []configv1.ClusterOperatorStatusCondition, condition *configv1.ClusterOperatorStatusCondition) []configv1.ClusterOperatorStatusCondition {
+	condition.LastTransitionTime = metav1.Now()
+
+	newConditions := []configv1.ClusterOperatorStatusCondition{}
+
+	found := false
+	for _, c := range oldConditions {
+		if condition.Type == c.Type {
+			if condition.Status == c.Status &&
+				condition.Reason == c.Reason &&
+				condition.Message == c.Message {
+				return oldConditions
+			}
+
+			found = true
+			newConditions = append(newConditions, *condition)
+		} else {
+			newConditions = append(newConditions, c)
+		}
+	}
+	if !found {
+		newConditions = append(newConditions, *condition)
+	}
+
+	return newConditions
+}
+
+// ConditionsEqual returns true if and only if the provided slices of conditions
+// (ignoring LastTransitionTime) are equal.
+func ConditionsEqual(oldConditions, newConditions []configv1.ClusterOperatorStatusCondition) bool {
+	if len(newConditions) != len(oldConditions) {
```

Comment:
Would this ever be true?

Reply:
Not sure what you're asking or what the problem is here.  Function or the `if` statement?  In any case, the function can return `true/false` and the `if` statement can evaluate to `true` or `false`.